CMakeLists.txt

```diff
@@ -45,21 +43,15 @@ mark_as_advanced(HIGHFIVE_USE_INSTALL_DEPS)
 # Check compiler cxx_std requirements
 # -----------------------------------
 
-if(NOT COMPILER_SUPPORTS_CXX11)
-  message(FATAL_ERROR "HighFive version >= 2.0 requires c++ standard >= c++11")
```

Comment:
I think this is a very useful check/info for users. Imagine the user uses this in another project with `CMAKE_CXX_STANDARD 98`. It will try to compile and give errors out of the blue.

Reply:
hm, I will look how others projects are doing it, but right now I was hopping the next few lines will handle it.

Reply:
While I don't think there was anything bad with the previous code I can agree with simplifying IF we provide similar functionality. I believe the only counter case was exactly the one I mentioned (`CMAKE_CXX_STANDARD being set to 98`). Should we 
```
IF (CMAKE_CXX_STANDARD EQUAL 98)
    message(FATAL_ERROR...
```